File: back-end/src/main/java/com/spring/restaurant/dto/PurchaseRequest.java
package com.spring.restaurant.dto;

import com.spring.restaurant.model.Address;
import com.spring.restaurant.model.Client;
import com.spring.restaurant.model.Item;
import com.spring.restaurant.model.RequestOrder;
import lombok.Data;


import java.util.List;

@Data
public class PurchaseRequest {

    private Client client;
    private RequestOrder requestOrder;
    private List<Item> items;
    private Address fromAddress;
    private Address toAddress;

}

File: front-end/src/app/model/purchase-request.ts
import {Client} from './client';
import {Address} from './address';
import {RequestOrder} from './request-order';
import {Item} from './item';

export class PurchaseRequest {
  client: Client;
  fromAddress: Address;
  toAddress: Address;
  requestOrder: RequestOrder;
  items: Item[];
}
